.github/workflows/go-mod-auto-bump.yml

```diff
@@ -8,6 +8,8 @@
 name: Go Mod Auto Version Update
 on: 
   push:
+    branches:
+    - ${{ github.ref }}
```

Comment:
I think with this syntax the workflow will run only when the push to `github.ref` is made. However, this variable is only accessible when the workflow is running, so, I suspect, that when GH Actions check if the condition for this workflow is met, it will see `github.ref` as a blank string, because as I said, this variable is only accessible when workflow is running. I will check if that is indeed like that.

Even if my suspicion is wrong, I do not really see why we are changing this. I do not see what is wrong with the "push to any branch" condition. Basically, the logic of this workflow is: when any branch `X` is changed, update go mod on this branch, and push the changes to `X` (aka `github.ref`)

Reply:
if I am right, I will check out other ways of how this can be done properly

Reply:
yes, this workflow will never run with these changes

Reply:
Why is that a problem?

This should be checked once a push is made to the current branch. A push triggers the workflow run where `github.ref` is accessible

Reply:
I understand now. ChatGPT was confusing me here:

![image](https://user-images.githubusercontent.com/34196718/226191749-2d5e8866-4045-4673-9c15-932a92f78eba.png)